homeassistant/components/shelly/config_flow.py

```diff
@@ -206,7 +207,7 @@ async def async_step_confirm_discovery(
         return self.async_show_form(
```

Comment:
Putting it here but should be in create entry -> do we want to store what generation it is? Or do you want to detect that each time on startup using get_info?

Reply:
I guess at startup is the best, as we support both.

Reply:
We do store generations, as this PR just aligns to `aioshelly`, I did not include extra functionality in it.
The idea to be implemented in the next PRs is:

1. For new devices a `gen` key will be added with a value of `1` or `2`
2. If key is missing we just setup a Gen1 device. We can update the key in config entry but it is not needed, it is safe to assume that if we don't find a key it is Gen1 device and if we find we take the generation from the key